### ISSUE TITLE
[JENKINS-35802] Developer can see running pipelines on the Pipeline summary views

### ISSUE THREAD

**Issue by jenkins-infra-bot** [jenkinsci/blueocean-plugin#2770]
<p>In Scope</p>
<ul>
	<li>Executing pipelines show with a blue circular indicator that slowly fills in</li>
	<li>Duration column should display the current time it has been running and update dynamically</li>
	<li>Completed column should contain an em dash <tt>–</tt></li>
	<li>Page should not need to be refreshed for new runs to appear</li>
	<li>Time ago should tick over (increase with time as the page is open)</li>
	<li>When a run finishes it should appear in the list with the correct status icon</li>
</ul>

---
<details><summary><i>Originally reported by <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/jamesdumay.png" title="jamesdumay's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=jamesdumay">jamesdumay</a>, imported from: <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-35802" target="_blank">Developer can see running pipelines on the Pipeline summary views</a></i></summary>
<i><ul>
<li><b>assignee</b>: <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/cliffmeyers.png" title="cliffmeyers's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=cliffmeyers">cliffmeyers</a>
<li><b>status</b>: Resolved
<li><b>priority</b>: Major
<li><b>component(s)</b>: blueocean-plugin
<li><b>label(s)</b>: blueocean-imported, japan, jira_escalated
<li><b>resolution</b>: Fixed
<li><b>resolved</b>: 2016-06-16T08:23:47+00:00
<li><b>votes</b>: 0
<li><b>watchers</b>: 0
<li><b>imported</b>: 2025-12-01
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p>In Scope</p>
<ul>
	<li>Executing pipelines show with a blue circular indicator that slowly fills in</li>
	<li>Duration column should display the current time it has been running and update dynamically</li>
	<li>Completed column should contain an em dash <tt>–</tt></li>
	<li>Page should not need to be refreshed for new runs to appear</li>
	<li>Time ago should tick over (increase with time as the page is open)</li>
	<li>When a run finishes it should appear in the list with the correct status icon</li>
</ul>
</pre>
</details>
</details>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/jamesdumay.png" title="jamesdumay's avatar" /> jamesdumay:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-35802?focusedId=261317&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-261317">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>This still needs another ticket for how we want to drive this from the backend.</p>
</pre>
</details>

</ul>
</details>

<p>This still needs another ticket for how we want to drive this from the backend.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/michaelneale.png" title="michaelneale's avatar" /> michaelneale:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-35802?focusedId=261492&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-261492">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>This is now unblocked. </p>

<p><a href="https://github.com/cloudbees/blueocean/tree/master/blueocean-rest#get-a-run-details" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/cloudbees/blueocean/tree/master/blueocean-rest#get-a-run-details</a> also has the data needed I believe to display percentage completion. </p>

<p>/run/x has: </p>
<div class="preformatted panel" style="border-width: 1px;"><div class="preformattedContent panelContent">
<pre>  "estimatedDurationInMillis" : 567,
..
  "startTime": "2016-03-16T09:02:26.498-0700",
}
</pre>
</div></div>

<p>completion-ness can be calculated from this: (current Time (browser) - startTime) / estimated</p>

<p>Is this reasonable for estimated for runs? we have SSE updating the rows dynamically for their state, so the updating of progress can ideally be done entirely client side (if it finishes on the server, the change will be pushed, so no need to refresh to get progress). 
<a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=jmcdonald" class="user-hover" rel="jmcdonald">jmcdonald</a> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=jdumay" class="user-hover" rel="jdumay">jdumay</a></p>
</pre>
</details>

</ul>
</details>

<p>This is now unblocked. </p>

<p><a href="https://github.com/cloudbees/blueocean/tree/master/blueocean-rest#get-a-run-details" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/cloudbees/blueocean/tree/master/blueocean-rest#get-a-run-details</a> also has the data needed I believe to display percentage completion. </p>

<p>/run/x has: </p>


```
  "estimatedDurationInMillis" : 567,
..
  "startTime": "2016-03-16T09:02:26.498-0700",
}

```

<p>completion-ness can be calculated from this: (current Time (browser) - startTime) / estimated</p>

<p>Is this reasonable for estimated for runs? we have SSE updating the rows dynamically for their state, so the updating of progress can ideally be done entirely client side (if it finishes on the server, the change will be pushed, so no need to refresh to get progress). <br/>
<a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=jmcdonald" class="user-hover" rel="jmcdonald">jmcdonald</a> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=jdumay" class="user-hover" rel="jdumay">jdumay</a></p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>sophistifunk:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-35802?focusedId=261442&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-261442">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><b>Important (in case I'm not the one doing this)</b>: Don't use `startTime`, use `durationInMillis`.</p>
</pre>
</details>

</ul>
</details>

<p><b>Important (in case I'm not the one doing this)</b>: Don't use `startTime`, use `durationInMillis`.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/jamesdumay.png" title="jamesdumay's avatar" /> jamesdumay:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-35802?focusedId=261412&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-261412">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=ataylor" class="user-hover" rel="ataylor">ataylor</a> this isn't the best ticket to link to for that Zendesk but <a href="https://issues.jenkins.io/browse/JENKINS-35849" title="Developer sees ETAs on hover states for all indicator types" class="issue-link" data-issue-key="JENKINS-35849"><del>UX-393</del></a> and UX-78 would be.</p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=ataylor" class="user-hover" rel="ataylor">ataylor</a> this isn't the best ticket to link to for that Zendesk but <a href="https://issues.jenkins.io/browse/JENKINS-35849" title="Developer sees ETAs on hover states for all indicator types" class="issue-link" data-issue-key="JENKINS-35849"><del>UX-393</del></a> and UX-78 would be.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/michaelneale.png" title="michaelneale's avatar" /> michaelneale:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-35802?focusedId=261475&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-261475">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=jmcdonald" class="user-hover" rel="jmcdonald">jmcdonald</a> when it is executing, you can't use durationInMillis as that means you have to fetch it each time to update (also, I am not sure if the duration is set until the run is resolved) - so do not thing that is right. </p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=jmcdonald" class="user-hover" rel="jmcdonald">jmcdonald</a> when it is executing, you can't use durationInMillis as that means you have to fetch it each time to update (also, I am not sure if the duration is set until the run is resolved) - so do not thing that is right. </p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>sophistifunk:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-35802?focusedId=261373&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-261373">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Just start counting locally once you get your result, and update it as SSEs come in. The results will be way more accurate than trusting that both the server and the client have their clocks set accurately.</p>
</pre>
</details>

</ul>
</details>

<p>Just start counting locally once you get your result, and update it as SSEs come in. The results will be way more accurate than trusting that both the server and the client have their clocks set accurately.</p>
